projects/js-packages/react-data-sync-client/src/DataSyncHooks.ts

```diff
@@ -12,7 +12,11 @@ import React from 'react';
 import { z } from 'zod';
 import { DataSync } from './DataSync';
 
-const queryClient = new QueryClient();
+/**
+ * @REACT-TODO This is temporary. We need to allow each app to define their own QueryClient.
+ * All of the functions below will have to be moved to a factory wrapper
+ */
+export const queryClient = new QueryClient();
```

Comment:
👍 I'm ok with this for now but would like to this on the board to make sure we take care of it before we call the migration done.

Before I add it to the board, a quick discussion: are we sure we need to? Will it cause any harm for DataSync to have its own queryClient that it exclusively works with, and for non-datasync REact Query stuff to live alongside it? I don't see a downside.

Reply:
It _may_ work but it's not correct. Think of queryClient as the namespace. Invalidation happens within a queryClient - so if I `invalidateAll`, I may accidentally invalidate applications I didn't intend to.

The fix is fairly simple - we'll just import `useDataSync` from boost, not from packages directly when we set this up - we can set those variables up in a lib directory statically. (something like `export const { useDataSync } = useDataSyncFactory()` )

Reply:
🤔 It's not a blocker here, but I don't understand. What is a scenario where we want to share a QueryClient with something that isn't DataSync?

Surely it'd be cleaner for DataSync to have its own cache, and any other use of ReactQuery to be kept separate?